python/pyarrow/tests/test_dataset.py

```diff
@@ -1577,6 +1577,38 @@ def test_dictionary_partitioning_outer_nulls_raises(tempdir):
         ds.write_dataset(table, tempdir, format='parquet', partitioning=part)
 
 
+def test_write_dataset_with_field_names(tempdir):
```

Comment:
I know the file is not super well structured (and there is a test just above that uses write_dataset as well), but _most_ write_dataset tests are below (after line 3000). So I would put this test for example right after `test_write_dataset_partitioned_dict(tempdir)`

Reply:
:+1: moved the tests